sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientImpl.java

```diff
@@ -0,0 +1,856 @@
+
+    private static final String APP_CONFIG_TRACING_NAMESPACE_VALUE = "Microsoft.AppConfiguration";
+
+    /**
+     * The proxy service used to perform REST calls.
+     */
+    private final ConfigurationService service;
+
+    /**
+     * Supported Cognitive Services endpoints (protocol and hostname, for example:
+     * https://westus2.api.cognitive.microsoft.com).
+     */
+    private final String endpoint;
+
+    /**
+     * Gets Supported Cognitive Services endpoints (protocol and hostname, for example:
+     * https://westus2.api.cognitive.microsoft.com).
+     *
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+    /**
+     * Api Version.
+     */
+    private final String apiVersion;
+
+    /**
+     * Gets Api Version.
+     *
+     * @return the apiVersion value.
+     */
+    public String getApiVersion() {
+        return this.apiVersion;
+    }
+
+    /**
+     * The HTTP pipeline to send requests through.
+     */
+    private final HttpPipeline httpPipeline;
+
+    /**
+     * Gets The HTTP pipeline to send requests through.
+     *
+     * @return the httpPipeline value.
+     */
+    public HttpPipeline getHttpPipeline() {
+        return this.httpPipeline;
+    }
+
+    /**
+     * The serializer to serialize an object into a string.
+     */
+    private final SerializerAdapter serializerAdapter;
+
+    private final ClientLogger logger = new ClientLogger(ConfigurationClientImpl.class);
+
+    /**
+     * Gets The serializer to serialize an object into a string.
+     *
+     * @return the serializerAdapter value.
+     */
+    public SerializerAdapter getSerializerAdapter() {
+        return this.serializerAdapter;
+    }
+
+    /**
+     * Initializes an instance of Configuration client.
+     *
+     * @param httpPipeline The HTTP pipeline to send requests through.
+     * @param serializerAdapter The serializer to serialize an object into a string.
+     * @param endpoint Supported Configuration Service endpoint.
+     * @param apiVersion Api Version.
+     */
+    public ConfigurationClientImpl(
+        HttpPipeline httpPipeline, SerializerAdapter serializerAdapter, String endpoint, String apiVersion) {
+        this.httpPipeline = httpPipeline;
+        this.serializerAdapter = serializerAdapter;
+        this.endpoint = endpoint;
+        this.apiVersion = apiVersion;
+        this.service =
+            RestProxy.create(ConfigurationService.class, this.getHttpPipeline(), this.getSerializerAdapter());
+    }
+
+    /**
+     * The interface defining all the services for {@link ConfigurationAsyncClient} to be used
+     * by the proxy service to perform REST calls.
+     * <p>
+     * This is package-private so that these REST calls are transparent to the user.
+     */
+    @Host("{url}")
+    @ServiceInterface(name = "AppConfig")
+    public interface ConfigurationService {
```

Comment:
Should still mark this as the service interface being used by the service clients